### DISCUSSION TITLE
d2j-dex2jar produced nothing

### DISCUSSION THREAD

**Discussion by haruame** [ThexXTURBOXx/dex2jar#50] [Q&A] [answered]
Hi @ThexXTURBOXx ,
I try to use d2j-dex2jar for the first time but nothing came out after these commands:
d2j-dex2jar.bat class.dex
sh d2j-dex2jar.sh -f class.dex
Please help me to correct the syntax if I'm doing it wrong!
Edit: sh d2j-dex2jar.sh -f ~/path/to/apk_to_decompile.apk but nothing came out either.
All ran with admin privilege Powershell/Git bash on Windows 11.

**Comment by ThexXTURBOXx** [comment]
Hi! Did you wait for long enough after running the command? Even though there is no output on the console, it might just take quite some time to finish.
Also, how big is your dex/apk?
Is there really no output on the console?

**Reply by haruame** [reply]
The APK is around 8MB, there are 2 dex files range from 7-9MB if I convert the apk to zip and extract the contents.
It ran like 1 second and return to the terminal
![image](https://github.com/ThexXTURBOXx/dex2jar/assets/50712531/1caf6132-d509-443b-ac73-8fed027b709c)
There's no extra log or files created in the folder.

**Reply by haruame** [reply]
Hi @ThexXTURBOXx ,
Could you please help?
If possible could I send the apk to you via email?

**Reply by ThexXTURBOXx** [reply]
Hi, I am working on this project in my free time - so I don't always have time to reply.
Feel free to send me the apk via email

**Reply by haruame** [reply]
Just sent you an email with the same title: d2j-dex2jar produced nothing

**Reply by ThexXTURBOXx** [reply, marked as answer]
@haruame Works for me just fine - even without Admin rights. Are you using the latest version (2.4.18)? Which Java (and version) do you have installed

![grafik](https://github.com/ThexXTURBOXx/dex2jar/assets/12763829/356cd6e0-1a11-437b-bbf3-7b925ac88066)


Converting just the `classes.dex` also works fine:

![grafik](https://github.com/ThexXTURBOXx/dex2jar/assets/12763829/dcb3280a-bfe1-46cc-b96d-64678cbd9e41)

**Reply by haruame** [reply]
hmmm, weird, I’m using Java 8 and latest version of dex2jar.
Does it need Java 21 to run?
I will install Java 21 and try to rerun it.
Thank you very much!

**Reply by ThexXTURBOXx** [reply]
Should also work with Java 8 - however if it does not work with Java 8, it's a bug and I will fix it :)
